react/tic-tac-toe/src/app/screens/Game/index.js

```diff
@@ -1,20 +1,80 @@
 import React from 'react';
 
+import { calculateWinner } from './components/utils';
 import styles from './styles.module.scss';
 import Board from './components/Board';
 
-function Game () {
-  return (
-    <div className={styles.game}>
-      <div className={styles.gameBoard}>
-        <Board />
-      </div>
-      <div className={styles.gameInfo}>
-        <div>{/* status */}</div>
-        <ol>{/* TODO */}</ol>
+class Game extends React.Component {
+  state = {
+    history: [{
+      squares: Array(9).fill(null)
+    }],
+    stepNumber: 0,
+    xIsNext: true,
+    winner: null
+  };
+
+  handleClick = (i) => {
+    const { history } = this.state;
+    const current = history[history.length - 1];
+    const squares = current.squares.slice();
+    if (this.state.winner || squares[i]) {
+      return;
+    }
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+    const winner = calculateWinner(squares);
+    this.setState({
+      winner,
+      history: history.concat([{
+        squares
+      }]),
+      stepNumber: history.length,
+      xIsNext: !this.state.xIsNext
+    });
+  }
+
+  jumpTo(step) {
+    this.setState({
+      stepNumber: step,
+      xIsNext: step % 2 === 0
+    });
+  }
+
+  render() {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[this.state.stepNumber];
+    const { winner } = this.state;
+    const moves = history.map((step, move) => {
+      const desc = move ?
+        'Go to move #' + move :
+        'Go to game start';
+      return (
+        <li key ={move}>
+          <button onClick={() => this.jumpTo(move)}>{desc}</button>
+        </li>
+      );
+    });
+
+
+    let status;
+    if (winner) {
+      status = 'Winner: ' + winner;
```

Comment:
The better form for do it i. You can read [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)
```
status = `Winner: ${winner}`;
```